app/models/miq_expression.rb

```diff
@@ -575,7 +575,7 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sql = Expression.build(@pexp, tz).to_sql if @pexp.present?
```

Comment:
I'm not sure what `@pexp` is here.  Also, it's not clear why we need to pass the timezone in.  From the looks of the classes below, some care about the tz, some don't.

Is it possible and does it makes sense to have something that looks like this (in terms of the timezone):

```ruby
if @pexp.present?
  exp_obj = Expression.build(@pexp.to_sql)
  exp_obj.timezone = tz if exp_obj.timezone_required?
  sql = exp_obj.to_sql
end
```

My suggestion above is quick and ugly, but my point is some of these things we're creating care about the tz, while others don't.  It seems we should ask them, maybe?  I don't know.

Note, naming is hard.  We have `@pexp`, `@exp` and the resulting objects of `.build` which are also `Expression-like`... Can we come up with better names? I don't remember this code so I have really no idea how they're all different.